pkg/kubelet/container/interface.go

```diff
@@ -0,0 +1,335 @@
+	Mounts []Mount
+	// Resources specifies the resource requirements for the sandbox.
+	// Note: On a Linux host, kubelet will create a pod-level cgroup and pass
+	// it as the cgroup parent for the PodSandbox. For some runtimes, this is
+	// sufficent to fulfill pod-level resource requirements, and this field
+	// will not be used. For others, e.g., hypervisor-based runtimes, explicit
+	// resource requirements for the sandbox are needed at creation time.
+	Resources PodSandboxResources
+	// Labels are key value pairs that may be used to scope and select individual resources.
+	Labels Labels
+	// Annotations is an unstructured key value map that may be set by external
+	// tools to store and retrieve arbitrary metadata.
+	Annotations map[string]string
+
+	// Linux contains configurations specific to Linux hosts.
+	Linux *LinuxPodSandboxConfig
+}
+
+// Labels are key value pairs that may be used to scope and select individual resources.
+// Label keys are of the form:
+//     label-key ::= prefixed-name | name
+//     prefixed-name ::= prefix '/' name
+//     prefix ::= DNS_SUBDOMAIN
+//     name ::= DNS_LABEL
+// The prefix is optional.  If the prefix is not specified, the key is assumed to be private
+// to the user.  Other system components that wish to use labels must specify a prefix.  The
+// "kubernetes.io/" prefix is reserved for use by kubernetes components.
+type Labels map[string]string
+
+// LinuxPodSandboxConfig holds platform-specific configuraions for Linux
+// host platforms and Linux-based containers.
+type LinuxPodSandboxConfig struct {
+	// CgroupParent is the parent cgroup of the sandbox.
+	CgroupParent string
+	// NamespaceOptions contains configurations for the sandbox's namespaces.
+	// This will be used only if the PodSandbox uses namespace for isolation.
+	NamespaceOptions NamespaceOptions
+}
+
+// NamespaceOptions provides options for Linux namespaces.
+type NamespaceOptions struct {
+	// HostNetwork uses the host's network namespace.
+	HostNetwork bool
+	// HostPID uses the host's pid namesapce.
+	HostPID bool
+	// HostIPC uses the host's ipc namespace.
+	HostIPC bool
+}
+
+// DNSOptions specifies the DNS servers and search domains.
+type DNSOptions struct {
+	// Servers is a list of DNS servers of the cluster.
+	Servers []string
+	// Searches is a list of DNS search domains of the cluster.
+	Searchs []string
+}
+
+type PodSandboxState string
+
+const (
+	// PodSandboxActive means the sandbox is functioning properly.
+	PodSandboxActive PodSandboxState = "active"
+	// PodSandboxInactive means the sandbox is not functioning properly.
+	PodSandboxInactive PodSandboxState = "inactive"
+)
+
+// PodSandboxFilter is used to filter a list of PodSandboxes.
+type PodSandboxFilter struct {
+	// Name of the sandbox.
+	Name string
+	// ID of the sandbox.
+	ID string
+	// State of the sandbox.
+	State string
+	// LabelSelector to select matches.
+	LabelSelector unversioned.LabelSelector
```

Comment:
I don't think we can support all the selector semantics in this API. It will be useful to clarify the specific groupings we will support, which are just `&&` and `==` for now.